File: freecodecamp/daily-code-challenge/2026-03-26.js
function getMovieNightCost(day, showtime, numberOfTickets) {
  // Step 1: Tuesday special
  if (day === "Tuesday") {
    return `$${(5 * numberOfTickets).toFixed(2)}`;
  }

  // Step 2: Base price
  let pricePerTicket;
  if (["Friday", "Saturday", "Sunday"].includes(day)) {
    pricePerTicket = 12;
  } else {
    pricePerTicket = 10;
  }

  // Step 3: Check matinee
  let time = showtime.slice(0, -2); // remove am/pm
  let period = showtime.slice(-2);

  let [hour, minute] = time.split(":").map(Number);

  if (period === "pm" && hour !== 12) hour += 12;
  if (period === "am" && hour === 12) hour = 0;

  let isMatinee = hour < 17; // before 5pm

  if (isMatinee) {
    pricePerTicket -= 2;
  }

  let total = pricePerTicket * numberOfTickets;

  return `$${total.toFixed(2)}`;
}

File: freecodecamp/daily-code-challenge/2026-03-26.py
def get_movie_night_cost(day, showtime, number_of_tickets):
    # Tuesday special
    if day == "Tuesday":
        return f"${5 * number_of_tickets:.2f}"

    # Base price
    if day in ["Friday", "Saturday", "Sunday"]:
        price = 12
    else:
        price = 10

    # Parse time
    time = showtime[:-2]
    period = showtime[-2:]

    hour, minute = map(int, time.split(":"))

    if period == "pm" and hour != 12:
        hour += 12
    if period == "am" and hour == 12:
        hour = 0

    # Matinee check
    if hour < 17:
        price -= 2

    total = price * number_of_tickets
    return f"${total:.2f}"